joomla/cassiopeia
# Use local Google Fonts

Pull Request for Issue #14 #15  .

### Summary of Changes
Added parameter(s) for GoogleFonts. 
Example fonts installed.
Removed Fira Sans

### Testing Instructions
Open template, you will see two new parameters
Google Font yes / no
Google Font Name as select field

![Screenshot_2020-08-12 Templates Edit Style - Joomla Cassiopeia Enhancement - Administration](https://user-images.githubusercontent.com/9153168/90011866-a2102080-dca2-11ea-87a0-a0f042d9c4e7.png)

### Expected result
If "no" is selected the template will use the font definitions from BS4 (at the moment)
If "yes" you can select a font or font combination. The selected font will be loaded with a specific css file.

![Screenshot_2020-08-12 Home](https://user-images.githubusercontent.com/9153168/90011894-ac321f00-dca2-11ea-949b-f6c6f6f79ba2.png)
![Screenshot_2020-08-12 Home-Josefin](https://user-images.githubusercontent.com/9153168/90011898-afc5a600-dca2-11ea-83ea-0596290a8713.png)
![Screenshot_2020-08-12 Home-nofont](https://user-images.githubusercontent.com/9153168/90011901-b3592d00-dca2-11ea-9d7b-6e9f750d109c.png)

### Documentation Changes Required

We need to explain how to install and use other fonts

#### Update 2020-08-21 by @richard67 

There are 2 use cases.

Case 1: CMS contibutors add new fonts to the sources so they are shipped with the CMS.

This should not really happen often since we should at the end ship only the fonts which are really used by the CMS, maybe a choice of 2 which look nice, but not a lot.

Upload a **s**css file for each font or combination of fonts and corresponding woff and woff2 files to folder `/build/media_source/fonts` and then run `npm ci`. That's it.

Compiled css and copied font files will be availabe in folder `/media/fonts` after that.

In the source code only what has changed or been added in folder `/build/media_source/fonts` has to be checked in.

Case 2: Users add new fonts to an installlled J4.

Upload a css file for each font or combination of fonts and corresponding woff and woff2 files to folder `/media/fonts`. Optionally you can also upload the minified css file in addition. That's it.

In both cases, after you have done the above steps (either case 1 or case 2), you can select in backend in Cassiopeia's template style settings the new fonts by the base name of the css files in the `/media/fonts` folder. Minified css files will be ignored.
